packages/nodejs/src/__tests__/config.test.ts

```diff
@@ -27,124 +54,211 @@ describe("Configuration", () => {
-        expect(config.sources.env).toEqual(overwrittenConfig)
+  describe("private environment variables", () => {
+    beforeEach(() => {
+      new Configuration({})
+    })
+
+    it("writes default configuration values to the environment", () => {
+      expect(env("_APPSIGNAL_ACTIVE")).toBeUndefined()
+      expect(env("_APPSIGNAL_APP_NAME")).toBeUndefined()
+      expect(env("_APPSIGNAL_CA_FILE_PATH")).toMatch(/cert\/cacert\.pem$/)
+      expect(env("_APPSIGNAL_DEBUG_LOGGING")).toBeUndefined()
+      expect(env("_APPSIGNAL_DNS_SERVERS")).toBeUndefined()
+      expect(env("_APPSIGNAL_ENABLE_HOST_METRICS")).toEqual("true")
+      expect(env("_APPSIGNAL_ENABLE_STATSD")).toBeUndefined()
+      expect(env("_APPSIGNAL_ENVIRONMENT")).toBeDefined()
+      expect(env("_APPSIGNAL_FILES_WORLD_ACCESSIBLE")).toEqual("true")
+      expect(env("_APPSIGNAL_FILTER_DATA_KEYS")).toBeUndefined()
+      expect(env("_APPSIGNAL_HOSTNAME")).toBeUndefined()
+      expect(env("_APPSIGNAL_HTTP_PROXY")).toBeUndefined()
+      expect(env("_APPSIGNAL_IGNORE_ACTIONS")).toBeUndefined()
+      expect(env("_APPSIGNAL_IGNORE_ERRORS")).toBeUndefined()
+      expect(env("_APPSIGNAL_IGNORE_NAMESPACES")).toBeUndefined()
+      expect(env("_APPSIGNAL_LOG")).toEqual("file")
+      expect(env("_APPSIGNAL_LOG_FILE_PATH")).toEqual("/tmp/appsignal.log")
+      expect(env("_APPSIGNAL_PUSH_API_ENDPOINT")).toEqual(
+        "https://push.appsignal.com"
+      )
+      expect(env("_APPSIGNAL_PUSH_API_KEY")).toBeUndefined()
+      expect(env("_APPSIGNAL_RUNNING_IN_CONTAINER")).toBeUndefined()
+      expect(env("_APPSIGNAL_TRANSACTION_DEBUG_MODE")).toBeUndefined()
+      expect(env("_APPSIGNAL_WORKING_DIRECTORY_PATH")).toBeUndefined()
+      expect(env("_APPSIGNAL_WORKING_DIR_PATH")).toBeUndefined()
+      expect(env("_APP_REVISION")).toBeUndefined()
+    })
+
+    it("writes private constants to the environment", () => {
+      expect(env("_APPSIGNAL_AGENT_PATH")).toMatch(/nodejs-ext\/ext$/)
+      expect(env("_APPSIGNAL_PROCESS_NAME")).toEqual("node")
+      expect(env("_APPSIGNAL_LANGUAGE_INTEGRATION_VERSION")).toEqual(
+        `nodejs-${VERSION}`
+      )
+      expect(env("_APPSIGNAL_APP_PATH")).toEqual(process.cwd())
+    })
+
+    describe("with config options set to non-default values", () => {
+      beforeEach(() => {
+        new Configuration({
+          name,
+          active: true,
+          apiKey,
+          debug: true,
+          dnsServers: ["8.8.8.8", "8.8.4.4"],
+          enableHostMetrics: false,
+          enableMinutelyProbes: false,
+          enableStatsd: true,
+          filesWorldAccessible: true,
+          filterDataKeys: ["password", "confirm_password"],
+          hostname: "MyHostName",
+          httpProxy: "http://localhost",
+          ignoreActions: ["MyAction", "MyOtherAction"],
+          ignoreErrors: ["MyError", "MyOtherError"],
+          ignoreNamespaces: ["MyNamespace", "MyOtherNamespace"],
+          logPath: os.tmpdir(),
```

Comment:
Nitpick: in Linux (that is, in our CI), this is `/tmp`, which _is_ the default value, so it's not testing non-default values.